### PR TITLE
Indy Helm Release template file generation role

### PR DESCRIPTION
**Pull request by himanshulalarya**
Signed-off-by: Lal <himanshu.a.lal@accenture.com>

### CODE REVIEW

File: platforms/hyperledger-indy/images/indy-cli/indy-ledger.sh [hunk 1]
```diff
@@ -1,54 +1,28 @@
...
-[ -z "$vault_addr" ] && { echo "Script Failed: Vault Address Missing"; exit 1;} || echo "Vault Address Received";
-[ -z "$vault_token" ] && { echo "Script Failed: Vault Token Missing"; exit 1;} || echo "VAult Token Received";
-[ -z "$admin_path" ] && { echo "Script Failed: Admin DID Path Missing"; exit 1;} || echo "Admin DID Path Received";
-[ -z "$admin_name" ] && { echo "Script Failed: Admin Name Missing"; exit 1;} || echo "Admin Name Received";
-[ -z "$identity_path" ] && { echo "Script Failed: Identity DID Path Missing"; exit 1;} || echo "Identity DID Received";
-[ -z "$identity_name" ] && { echo "Script Failed: Identity DID Name Missing"; exit 1;} || echo "Identity Name Received";
+admin_did=$1
+admin_seed=$2
+identity_did=$3
+identity_seed=$4
```
Comment (juraj-durech) [inline platforms/hyperledger-indy/images/indy-cli/indy-ledger.sh]: No identity_seed to be passed as the admin identity and new identity may come from different organisations. The job itself will run in the K8S cluster of the admin organisation thus just the private crypto of the admin might be used.

File: platforms/hyperledger-indy/configuration/roles/create/helm_component/ledger_txn/tasks/identity.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#----------------------------------------------------------------------  
+    # create deployment file for node steward
+    - name: "Create new identity ledger transaction deployment file"
+      include_role:
+        name: ledger-tpl
```
Comment (juraj-durech) [inline platforms/hyperledger-indy/configuration/roles/create/helm_component/ledger_txn/tasks/identity.yaml]: What's this role please?

File: platforms/hyperledger-indy/configuration/roles/create/helm_component/ledger_txn/tasks/identity.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+#----------------------------------------------------------------------
```
Comment (juraj-durech) [inline platforms/hyperledger-indy/configuration/roles/create/helm_component/ledger_txn/tasks/identity.yaml]: This task should add just ONE identity assuming that the admin identity and new identity come from different organisations. The logic which chooses the admin identity (and admin organisation) and the new identity (with its organisation) should be in the setup role which would call this one.

File: platforms/hyperledger-indy/configuration/roles/create/helm_component/ledger_txn/tasks/identity.yaml [hunk 4]
```diff
@@ -0,0 +1,19 @@
+#----------------------------------------------------------------------  
+    # create deployment file for node steward
+    - name: "Create new identity ledger transaction deployment file"
+      include_role:
+        name: ledger-tpl
+      vars:
+        component_type: "identity"
+        component_name: "{{ organizationItem.name }}"
+        indy_version: "indy-{{ network.version }}"
+        release_dir: "/home/endorsers"
```
Comment (juraj-durech) [inline platforms/hyperledger-indy/configuration/roles/create/helm_component/ledger_txn/tasks/identity.yaml]: Probably bad path

File: platforms/hyperledger-indy/configuration/roles/create/helm_component/ledger_txn/tasks/nested_main.yaml [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+# This task tests the value file for syntax errors/ missing values
+# This is done by calling the helm_lint role and passing the value file parameter
+# When a new k8_component is added, changes should be made in helm_lint role as well
+- name: Helm lint
+  include_role:
+    name: "{{ playbook_dir }}/../../shared/configuration/roles/helm_lint"
+  vars:
+    helmtemplate_type: "{{ component_type }}"
+    chart_path: "{{ gitops.chart_source }}"
+    value_file: "{{ release_dir }}/{{ component_name }}/{{ endorser_item }}.yaml"
```
Comment (juraj-durech) [inline platforms/hyperledger-indy/configuration/roles/create/helm_component/ledger_txn/tasks/nested_main.yaml]: newline missing

File: platforms/hyperledger-indy/configuration/roles/create/helm_component/ledger_txn/tasks/main.yaml [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+  file:
+    path: "{{ component_name }}"
+    recurse: yes
+    state: directory
+
+- name: Create HelmRelease file
+  include_tasks: nested_main.yaml
+  vars:
+    endorser_item: "{{ newIdentityItem.name }}"
+    admin_name: "{{ component_name }}-trustee"
```
Comment (juraj-durech) [inline platforms/hyperledger-indy/configuration/roles/create/helm_component/ledger_txn/tasks/main.yaml]: Name should come from network.yaml

File: platforms/hyperledger-indy/configuration/roles/create/helm_component/ledger_txn/tasks/nested_main.yaml [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+##############################################################################################
+# This task creates deployment file for new identities.
+- name: "create value file for {{ component_name }} {{ component_type }}"
+  template:
+    src: "{{ dlt_templates[component_type] }}"
+    dest: "{{ values_file }}"
+  vars:
+    values_file: "{{ release_dir }}/{{ component_name }}/{{ endorser_item }}-{{ component_type }}.yaml"
+    chart: "indy-ledger-txn"
+    newIdentity_role: "ENDORSER"
```
Comment (juraj-durech) [inline platforms/hyperledger-indy/configuration/roles/create/helm_component/ledger_txn/tasks/nested_main.yaml]: Should be passed on from the new setup role.

File: platforms/hyperledger-indy/configuration/roles/create/helm_component/ledger_txn/tasks/identity.yaml (hunk 2 above)
Comment (himanshulalarya) [reply platforms/hyperledger-indy/configuration/roles/create/helm_component/ledger_txn/tasks/identity.yaml]: This role is used for running the helm release role .(as a starting point)

Comment (juraj-durech) [reply platforms/hyperledger-indy/configuration/roles/create/helm_component/ledger_txn/tasks/identity.yaml]: And where is it located in the Git repo?

File: platforms/hyperledger-indy/configuration/roles/create/helm_component/ledger_txn/tasks/nested_main.yaml [hunk 8]
```diff
@@ -0,0 +1,69 @@
+################################################################################################
```
Comment (juraj-durech) [inline platforms/hyperledger-indy/configuration/roles/create/helm_component/ledger_txn/tasks/nested_main.yaml]: You use component_name as the name for both organisations (admin org and endorser org), these can be different orgs. Same with org_vault_url...

Comment (himanshulalarya) [reply platforms/hyperledger-indy/configuration/roles/create/helm_component/ledger_txn/tasks/nested_main.yaml]: Changed the names, now different for both admin and new identities

**Pull request merged**